src/App/Command/GithubCQRSEndpointsCommand.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Console\App\Command;
+
+use Console\App\Service\Github;
+use Github\Api\Search;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Helper\TableCell;
+use Symfony\Component\Console\Helper\TableSeparator;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class GithubCQRSEndpointsCommand extends Command
+{
+    /**
+     * @var Github;
+     */
+    protected $github;
+
+    const NUMBER_OF_ITEMS_PER_PAGE = 100;
+    const ADAPTER_REPOSITORY_PATH = 'src/Adapter/';
+    const COMMAND_HANDLER = 'CommandHandler';
+    const QUERY_HANDLER = 'QueryHandler';
+    const HANDLER_SUFFIX = 'Handler';
+
+    protected function configure()
+    {
+        $this->setName('github:cqrs:endpoints')
+            ->setDescription('Get list of cqrs endpoints')
+            ->addOption(
+                'ghtoken',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                '',
+                $_ENV['GH_TOKEN'] ?? null
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->github = new Github($input->getOption('ghtoken'));
+        $time = time();
+        $this->getCQRSEndpoints($input, $output);
+        $output->writeLn(['', 'Output generated in ' . (time() - $time) . 's.']);
+
+        return 0;
+    }
+
+    private function getCQRSEndpoints(InputInterface $input, OutputInterface $output)
+    {
+        $rows[] = [new TableCell('<info> CQRS Endpoints</info>', ['colspan' => 2])];
+        $rows[] = new TableSeparator();
+        $files = [];
+        $query = 'filename:*Handler.php+repo:Prestashop/Prestashop';
+
+        $api = new Search($this->github->getClient());
+        $api->setPerPage(self::NUMBER_OF_ITEMS_PER_PAGE);
+        $handlers = $api->code($query);
+        $currentItem = 0;
+        $selectedItems = 0;
+        for ($i = 0; $i < $handlers['total_count'] / self::NUMBER_OF_ITEMS_PER_PAGE; ++$i) {
+            foreach ($handlers['items'] as $handler) {
+                ++$currentItem;
+                $pathName = pathinfo(substr($handler['path'], strlen(self::ADAPTER_REPOSITORY_PATH)))['dirname'];
+                if (str_starts_with($handler['path'], self::ADAPTER_REPOSITORY_PATH)) {
+                    if (!isset($files[$pathName])) {
+                        $files[$pathName] = [];
+                    }
+                    ++$selectedItems;
+                    $files[$pathName][$handler['name']] = [
+                        'id' => $currentItem,
+                        'name' => $handler['name'],
+                    ];
+                }
+            }
+            // it stinks, but needed to not exceed quota on github API (https://docs.github.com/en/rest/rate-limit)
+            sleep(7);
```

Comment:
Needs to be in constant API_CALL_DELAY amybe ?

Reply:
you're also right .....I'm a spider-pig !

Reply:
I can also walk on the ceiling